CMakeLists.txt

```diff
@@ -213,12 +217,16 @@ if(BML_MPI)
   add_definitions(-DDO_MPI)
 endif()
 
-include(FindCUDAToolkit)
-if(CUDAToolkit_FOUND)
-  message("Found CUDAToolkit: ${CUDAToolkit_TARGET_DIR}")
-endif()  
+if(BML_CUDA OR BML_CUSOLVER OR BML_CUSPARSE)
+  include(FindCUDAToolkit)
+  if(CUDAToolkit_FOUND)
+    message("Found CUDAToolkit: ${CUDAToolkit_TARGET_DIR}")
+  endif()
+endif()
 
-find_package(hip QUIET)
+if(BML_ROCSOLVER OR BML_ROCSPARSE)
```

Comment:
What about BML_MAGMA?

Reply:
It looks like BML_MAGMA might be needed here, the following happens when building against MAGMA on crusher:

/sw/crusher/spack-envs/base/opt/cray-sles15-zen3/gcc-11.2.0/magma-2.6.1-4ns6vbbhaeosp36tm4jvl3e3ohxxrzpb/include/magma_types.h:148:14: fatal error: hipblas.h: No such file or directory
  148 |     #include <hipblas.h>

Reply:
Ok, looks like magma is installed differently on crusher. Added BML_MAGMA flag

Reply:
Thanks, that worked.